Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -195,10 +199,79 @@ private MapPolyline parseLineString()
         skipToEndOfTag();
       }
     }
-    verifyCoordinatesParsed(hasParsedCoordinates);
+    verifyElementSeen("coordinates", hasParsedCoordinates);
     return line;
   }
 
+  @NonNull
+  private MapPolygon parsePolygon() throws IOException, XmlPullParserException, KMLParseException {
+    mParser.require(XmlPullParser.START_TAG, mNameSpace, "Polygon");
+    MapPolygon polygon = mFactory.createMapPolygon();
+    ArrayList<ArrayList<Geoposition>> rings = new ArrayList<>();
+    boolean hasOuterBoundary = false;
+    AltitudeReferenceSystemWrapper altitudeReferenceSystemWrapper =
+        new AltitudeReferenceSystemWrapper(AltitudeReferenceSystem.GEOID);
+    while (mParser.next() != XmlPullParser.END_TAG) {
+      if (mParser.getEventType() != XmlPullParser.START_TAG) {
+        continue;
+      }
+      String type = mParser.getName();
+      if (type.equals("outerBoundaryIs") || type.equals("innerBoundaryIs")) {
+        if (type.equals("outerBoundaryIs")) {
+          verifyElementNotSeen("outerBoundaryIs", hasOuterBoundary);
+          hasOuterBoundary = true;
+        }
+        ArrayList<Geoposition> positions = parsePolygonRing(type, altitudeReferenceSystemWrapper);
```

Comment:
No need for named local variable, can just pass parsePolygonRing result to rings.add()